src/components/Form/Form.js

```diff
@@ -12,19 +14,20 @@ export default function Form({ editMode, exitEditMode, id }) {
 		if (editMode) {
 			const task = tasks.find(element => element.id === id);
 			setInputValue(task.name);
+			setDueDateValue(task.dueDate);
 		}
 	}, []);
 
-	function onSubmit(inputValue, event) {
-		event.preventDefault();
+	function onSubmit(event, inputValue, dueDateValue) {
 		if (tasks.some(task => task.name === inputValue && task.id !== id)) {
 			window.confirm('Task already in the list');
 		} else {
 			if (editMode) {
-				updateTask(id, inputValue);
+				updateTask(id, inputValue, dueDateValue);
 				exitEditMode();
 			} else {
-				addTask(inputValue);
+				event.preventDefault();
+				addTask(inputValue, dueDateValue);
 				setInputValue('');
```

Comment:
I think you forgot to add here `setDueDateValue('');` so that you can see the placeholder again after submit :)

Reply:
Good point, Thanks. I will try to implement this before merging.